File: app/src/main/java/com/boardgamegeek/util/fabric/PlayerColorsManipulationEvent.java
package com.boardgamegeek.util.fabric;

import com.crashlytics.android.answers.Answers;

public class PlayerColorsManipulationEvent extends DataManipulationEvent {
	public static void log(String action) {
		PlayerColorsManipulationEvent event = new PlayerColorsManipulationEvent(action);
		Answers.getInstance().logCustom(event);
	}

	public static void log(String action, String color) {
		PlayerColorsManipulationEvent event = new PlayerColorsManipulationEvent(action);
		event.putCustomAttribute("Color", color);
		Answers.getInstance().logCustom(event);
	}

	protected PlayerColorsManipulationEvent(String action) {
		super("PlayerColors");
		putCustomAttribute("action", action);
	}
}

File: app/src/main/java/com/boardgamegeek/util/fabric/PlayerColorsManipulationEvent.kt
package com.boardgamegeek.util.fabric

import com.crashlytics.android.answers.Answers

class PlayerColorsManipulationEvent(action: String) : DataManipulationEvent("PlayerColors") {

    init {
        putCustomAttribute("action", action)
    }

    companion object {
        fun log(action: String) {
            val event = PlayerColorsManipulationEvent(action)
            Answers.getInstance().logCustom(event)
        }

        fun log(action: String, color: String) {
            val event = PlayerColorsManipulationEvent(action).apply {
                putCustomAttribute("Color", color)
            }
            Answers.getInstance().logCustom(event)
        }
    }
}
